jlm/hls/ir/hls.hpp

```diff
@@ -100,14 +138,38 @@ class fork_op final : public jlm::rvsdg::simple_op
     return std::unique_ptr<jlm::rvsdg::operation>(new fork_op(*this));
   }
 
+  /**
+   * Create a fork operation with a single input and multiple outputs.
+   *
+   * /param nalternatives Number of outputs.
+   * /param value The signal type, which is the same for the input and all outputs.
+   * /param isConstant If true, the fork is a constant fork.
+   *
+   * /return A vector of outputs.
+   */
   static std::vector<jlm::rvsdg::output *>
-  create(size_t nalternatives, jlm::rvsdg::output & value)
+  create(size_t nalternatives, jlm::rvsdg::output & value, bool isConstant = false)
   {
 
     auto region = value.region();
-    fork_op op(nalternatives, value.Type());
+    fork_op op(nalternatives, value.Type(), isConstant);
     return jlm::rvsdg::simple_node::create_normalized(region, op, { &value });
   }
+
+  /**
+   * Cechk if a fork is a constant fork (CFORK).
+   *
+   * /return True if the fork is a constant fork, i.e., the input of the fork is a constant, else
+   * false.
+   */
+  bool
+  IsConstant() const
+  {
+    return IsConstant_;
+  }
+
+private:
+  bool IsConstant_ = false;
```

Comment:
For the future: I am not sure whether there is a particular reason that I a do not see as to why this property is here, but it might be better to implement this when you lower the fork operation. It should be trivial to determine whether a constant is connected to a fork or not. It is  a simple tracing of the fork input to the next node and then checking whether the node is a constant or not.

Reply:
Agreed, the minor benefit of having this could be during debug or viewing the graph as for example the debug_string() depends on it.
Could change IsConstant() to check the inputs of the source node.